pkg/controllers/addons_controller.go

```diff
@@ -270,30 +163,30 @@ func (r *AddonsLayerReconciler) update(ctx context.Context, log logr.Logger,
 }
 
 /*
-func (r *AddonsLayerReconciler) gitRepositorySource(o handler.MapObject) []ctrl.Request {
+func (r *AddonsLayerReconciler) sourceController(o handler.MapObject) []ctrl.Request {
 	//ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
 	//defer cancel()
 
 	return []ctrl.Request{
 		{
 			NamespacedName: types.NamespacedName{
-				//Name:      sourcev1.GitRepository.Name,
-				//Namespace: sourcev1.GitRepository.Namespace,
+				Name:      sourcev1.GitRepository.Name,
+				Namespace: sourcev1.GitRepository.Namespace,
 			},
 		},
 	}
 }
 */
-
 // SetupWithManager is used to setup the controller
 func (r *AddonsLayerReconciler) SetupWithManager(mgr ctrl.Manager) error {
```

Comment:
@paulcarlton-ww It might be better to introduce a config options struct called AddonsLayerReconcilerOptions. This can have "MaxConcurrentReconciles" option which can then be used to pass on to "		WithOptions(controller.Options{MaxConcurrentReconciles: opts.MaxConcurrentReconciles}) while building the controller into the mgr using ctrl.NewControllerManagedBy
". This will allow the main func to retrieve the value for concurrent reconciling from the user and set it while bootstrapping into the manager. This config options struct can also be a container for us to expose more configuration options in the future.

Similar to https://github.com/fluxcd/kustomize-controller/blob/master/controllers/kustomization_controller.go#L219

Reply:
https://github.com/fidelity/kraan/issues/17 to capture this task